src/components/pages/manage/movies/create.tsx

```diff
@@ -1,91 +1,156 @@
+import { Title } from '@/ui/Title';
 import { getApplicationServerSideProps } from '@/utils/getApplicationServerSideProps';
 
 const getValue = getValueFromTheme('moviesCreatePage');
 
-const Step1Content = () => {
-  const [movieState, sendMovieEvent] = useMachine(movieMachine);
+type StepProps = StackProps & { step: number };
+
+type MachineProps = {
+  movieState: State<MovieContext, MovieEvent>;
+  sendMovieEvent: (event: MovieEvent) => State<MovieContext, MovieEvent>;
+};
 
+const Step = ({ step, children, ...props }: StepProps) => {
   const { t } = useTranslation();
 
-  const handleSelectTheme = (value: Values<typeof MovieThemes>) => {
-    sendMovieEvent(MovieEventTypes.CHOOSE_THEME, { value });
-  };
+  return (
+    <Stack spacing={4} {...getStackProps(props)}>
+      <Title
+        color={getValue('title.color')}
+        lineHeight="220%"
+        alignItems="center"
+        spacing={3}
+        css={`
+          border-bottom: 1px solid ${getValue('title.borderColor')};
+        `}
+      >
+        <Box
+          borderRadius="50%"
+          width="1.8rem"
+          height="1.8rem"
+          lineHeight="1.8rem"
+          backgroundColor={getValue('stepNumber.backgroundColor')}
+          padding={0}
+          fontSize="1rem"
+          fontWeight="bold"
+          color="white"
+          textAlign="center"
+        >
+          {step}
+        </Box>
+        <Text>{t(`movies.create.step${step}_title`)}</Text>
+      </Title>
+      {children}
+    </Stack>
+  );
+};
+
+type Step1ContentProps = StackProps & MachineProps;
 
-  const handleClearTheme = () => sendMovieEvent(MovieEventTypes.CLEAR_THEME);
+const Step1Content = ({
+  movieState,
+  sendMovieEvent,
+  ...props
+}: Step1ContentProps) => {
+  const { t } = useTranslation();
 
   return (
-    <Inline spacing={3} flexWrap="wrap" maxWidth="70rem">
-      {movieState.context.theme === null ? (
-        Object.entries(MovieThemes).map(([key, value]) => (
-          <Button
-            width="auto"
-            marginBottom={3}
-            display="inline-flex"
-            key={key}
-            variant={ButtonVariants.SECONDARY}
-            onClick={() => handleSelectTheme(value)}
-          >
-            {t(`themes*${value}`, { keySeparator: '*' })}
-          </Button>
-        ))
-      ) : (
-        <Inline alignItems="center" spacing={3}>
-          <Icon
-            name={Icons.CHECK_CIRCLE}
-            color={getValue('check.circleFillColor')}
-          />
-          <Text>
-            {t(`themes*${movieState.context.theme}`, { keySeparator: '*' })}
-          </Text>
-          <Button variant={ButtonVariants.LINK} onClick={handleClearTheme}>
-            {t('movies.create.actions.change_theme')}
-          </Button>
-        </Inline>
-      )}
-    </Inline>
+    <Stack spacing={5} {...getStackProps(props)}>
+      <Inline spacing={3} flexWrap="wrap" maxWidth="70rem">
+        {movieState.context.theme === null ? (
```

Comment:
not sure if this is the preferred way to do this.
seems a bit weird to create designated states and then write ternaries based on values in the context

Reply:
It's probably better to base theis logic on the current state it's in, but right now we only have the idle state since we simplified the movie state machine